## Bases de Dados - Parte 2

Este notebook tem por objeto explorar mais funcionalidades em bases de dados.

Vamos explorar uma base de dados de IDH ...
...
explicar mais a base de dados

Para reduzir o tamanho do arquivo vamos utilizar apenas os dados coletados em 1991

In [ ]:
import pandas as pd

In [15]:
idh = pd.read_csv("https://drive.google.com/u/3/uc?id=1hbLQ68EhxYUjsi2IaGzO59EW_r3CKVgp&export=download", sep=";")


In [14]:
idh

REALIZACAO  CODIGO  UF REGIAO  ... POP_10AN  POP_7AN TX_MRT RD_FM
0           1991     190  AP      N  ...      NaN      NaN    NaN   NaN
1           1991      18  AC      N  ...      NaN      NaN    NaN   NaN
2           1991      36  AL     NE  ...      NaN      NaN    NaN   NaN
3           1991      50  AL     NE  ...      NaN      NaN    NaN   NaN
4           1991      90  AL     NE  ...      NaN      NaN    NaN   NaN
...          ...     ...  ..    ...  ...      ...      ...    ...   ...
5503        1991    4170  RS      S  ...    432.0    758.0    5,9  1,45
5504        1991    4470  SC      S  ...   3037.0   5014.0   7,44  1,46
5505        1991    5269  SP     SE  ...  37275.0  59664.0  20,96  3,05
5506        1991    4444  SC      S  ...  25416.0  41652.0  24,63  2,73
5507        1991    4043  RS      S  ...   1267.0   2146.0    5,9  1,43

[5508 rows x 17 columns]

In [13]:
idh.head()

REALIZACAO  CODIGO  UF REGIAO  ... POP_10AN POP_7AN TX_MRT RD_FM
0        1991     190  AP      N  ...      NaN     NaN    NaN   NaN
1        1991      18  AC      N  ...      NaN     NaN    NaN   NaN
2        1991      36  AL     NE  ...      NaN     NaN    NaN   NaN
3        1991      50  AL     NE  ...      NaN     NaN    NaN   NaN
4        1991      90  AL     NE  ...      NaN     NaN    NaN   NaN

[5 rows x 17 columns]

Use o método .info () para verificar quantas linhas existem na tabela (arquivo) e também quais são os atributos (colunas).

In [16]:
idh.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5508 entries, 0 to 5507
Data columns (total 17 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   REALIZACAO  5508 non-null   int64  
 1   CODIGO      5508 non-null   int64  
 2   UF          5508 non-null   object 
 3   REGIAO      5508 non-null   object 
 4   MUNICIPIO   5508 non-null   object 
 5   LONGVD      4491 non-null   object 
 6   EDUC        4491 non-null   object 
 7   RENDA       4491 non-null   object 
 8   ESP_VD_NC   4491 non-null   object 
 9   POP_URB     4491 non-null   object 
 10  IDHM        4491 non-null   object 
 11  POP_25AN    4491 non-null   float64
 12  POP_15AN    4491 non-null   float64
 13  POP_10AN    4491 non-null   float64
 14  POP_7AN     4491 non-null   float64
 15  TX_MRT      4491 non-null   object 
 16  RD_FM       4491 non-null   object 
dtypes: float64(4), int64(2), object(11)
memory usage: 731.7+ KB


#Dados ausentes
Métodos convenientes para lidar com Missing Data em pandas:

O método "dropna" elimina todas as linhas que tenham algum atributo com valor vazio

In [17]:
idh_aux = idh.dropna()

Note que ao verificar as novas informações da tabela com info() idh_aux tem 4491 linhas (antes do drop eram 5508 entradas).

In [21]:
idh_aux.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4491 entries, 1017 to 5507
Data columns (total 17 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   REALIZACAO  4491 non-null   int64  
 1   CODIGO      4491 non-null   int64  
 2   UF          4491 non-null   object 
 3   REGIAO      4491 non-null   object 
 4   MUNICIPIO   4491 non-null   object 
 5   LONGVD      4491 non-null   object 
 6   EDUC        4491 non-null   object 
 7   RENDA       4491 non-null   object 
 8   ESP_VD_NC   4491 non-null   object 
 9   POP_URB     4491 non-null   object 
 10  IDHM        4491 non-null   object 
 11  POP_25AN    4491 non-null   float64
 12  POP_15AN    4491 non-null   float64
 13  POP_10AN    4491 non-null   float64
 14  POP_7AN     4491 non-null   float64
 15  TX_MRT      4491 non-null   object 
 16  RD_FM       4491 non-null   object 
dtypes: float64(4), int64(2), object(11)
memory usage: 631.5+ KB


#Preenchendo vazios com valores
Também é possível substituir os valores VAZIOS por outros valores automaticamente. Para tal, pode-se usar o método "fillna"

In [23]:
idh.fillna(value=0)

REALIZACAO  CODIGO  UF REGIAO  ... POP_10AN  POP_7AN TX_MRT RD_FM
0           1991     190  AP      N  ...      0.0      0.0      0     0
1           1991      18  AC      N  ...      0.0      0.0      0     0
2           1991      36  AL     NE  ...      0.0      0.0      0     0
3           1991      50  AL     NE  ...      0.0      0.0      0     0
4           1991      90  AL     NE  ...      0.0      0.0      0     0
...          ...     ...  ..    ...  ...      ...      ...    ...   ...
5503        1991    4170  RS      S  ...    432.0    758.0    5,9  1,45
5504        1991    4470  SC      S  ...   3037.0   5014.0   7,44  1,46
5505        1991    5269  SP     SE  ...  37275.0  59664.0  20,96  3,05
5506        1991    4444  SC      S  ...  25416.0  41652.0  24,63  2,73
5507        1991    4043  RS      S  ...   1267.0   2146.0    5,9  1,43

[5508 rows x 17 columns]